### PR TITLE
custom header support

### LINKED ISSUE #195: Configurable health response options via middleware?
Currently I'm using `terminus` in conjunction with `express`. I'm using the `cors` middleware.

I have a React frontend that occasionally hits the `/health` endpoint generated by `terminus` and tracks whether or not the API is healthy.

However, the health endpoints generated by `terminus` do not set any CORS headers or allow any middleware to be used: https://github.com/godaddy/terminus/blob/7eb6efaeef910d0b5bdae79c4e70474cbfde1b10/lib/terminus.js#L82-L94

So my frontend cannot make a request to the health endpoint.

For now my solution is to just not use `terminus`' healthchecks, but I wonder if there is a way to do this that I just haven't found yet?

### PR DESCRIPTION
**Pull request by LuigiZaccagnini**
fixes #195 

- Added changes to allow for headers to be written
- I also updated the documents to show how to include headers to be written. 

An example on how to write headers would be: 
```js
const http = require("http");
const express = require("express");
const terminus = require("../terminus/index.js");
const app = express();

app.get("/", (req, res) => {
  res.send("ok");
});

const server = http.createServer(app);

function healthCheck({ state }) {
  // `state.isShuttingDown` (boolean) shows whether the server is shutting down or not
  return Promise
    .resolve
    // optionally include a resolve value to be included as
    // info in the health check response
    ();
}

const options = {
  healthChecks: {
    "/healthcheck": healthCheck,
    verbatim: true,
    __unsafeExposeStackTraces: true,
  },
  headers: {
    "Access-Control-Allow-Origin": "*",
    "Access-Control-Allow-Methods": "OPTIONS, POST, GET",
  },
};

terminus.createTerminus(server, options);

server.listen(3000);
```

I tested it to enable CORS headers and everything seems to be working!

### CODE REVIEW

File: lib/terminus.js [hunk 1]
```diff
@@ -72,7 +74,7 @@ const intialState = {
 
 function noop () {}
 
-function decorateWithHealthCheck (server, state, options) {
+function decorateWithHealthCheck (server, state, options, headers) {
```
Comment (gergelyke-dd) [inline lib/terminus.js:77]: should `headers` be part of `options` internally too?

**Comment by ste2425** [comment]
Is there any movement on this? We've come up against a CORS issue that this functionality would allow us to solve. Would be great if this could get in.

**Comment by LuigiZaccagnini** [comment]
@ste2425 Going to resolve the requested change today!

**Comment by LuigiZaccagnini** [comment]
Hey @gergelyke-dd! I can't re-request a review so I just wanted to tag you to tell you I finished the changes!

**Comment by gergelyke** [comment]
Looks great! One question: with the introduction of res.writeHead, do we still need the res.status = … lines?

**Comment by LuigiZaccagnini** [comment]
@gergelyke good call! Just removed them and tested. Everything still works and the res.status is gone!

**Comment by gergelyke** [comment]
one last thing before we land it , would you mind adding some test coverage?

**Comment by LuigiZaccagnini** [comment]
@gergelyke just added test coverage!

**Comment by gergelyke** [comment]
awesome, thank you 👏

**Pull request merged**